docs/examples/cpp.md

```diff
@@ -0,0 +1,115 @@
+# C++ example
+
+In this example, we will walk through a simple C++ app that uses CMake.
+
+Let's assume our code is structured as follows:
+```
+.
+├── build.earth
+└── src
+    ├── CMakeLists.txt
+    ├── fib.cpp
+    ├── fib.h
+    └── main.cpp
+```
+
+Our program will be split between two different cpp files; the main.cpp file:
+
+```C++
+#include <iostream>
+
+#include "fib.h"
+
+int main(int argc, char *argv[]) {
+	for( int i = 0; i < 5; i++ ) {
+		std::cout << "fib(" << i << ") = " << fib(i) << std::endl;
+	}
+	return 0;
+}
+```
+
+and a file containing our fibonacci function:
+
+```C++
+#include "fib.h"
+
+int fib(int n)
+{
+	if( n <= 0 ) {
+		return 0;
+	}
+	if( n == 1 ) {
+		return 1;
+	}
+	return fib(n-1) + fib(n-2);
+}
+```
+
+We will use CMake to manage the build process of the c++ code, with the following CMakeList.txt file:
+
+```
+cmake_minimum_required(VERSION 2.8.9)
+project (fibonacci)
+add_executable(fibonacci main.cpp fib.cpp)
+```
+
+Cmake caches object files under CmakeFiles which allows cmake to only recompile objects when the corresponding
+source code changes. We will use a mount-based cache to cache these temporary files to allow for faster builds
```

Comment:
Might make sense to link `mount-based cache` to some page that describes a bit more what that is. Either the Earthfile reference, or the Advanced caching techniques page.